route53-record.tf

```diff
@@ -1,13 +1,31 @@
 data "aws_route53_zone" "selected" {
   count = var.alb_only && var.hostname_create ? 1 : 0
   name  = var.hosted_zone
+  private_zone = var.hosted_zone_is_internal
 }
 
 resource "aws_route53_record" "hostnames" {
-  count   = var.alb_only && var.hostname_create && length(var.hostnames) != 0 ? length(var.hostnames) : 0
+  count   = !var.hosted_zone_is_internal && var.alb_only && var.hostname_create && length(var.hostnames) != 0 ? length(var.hostnames) : 0
   zone_id = data.aws_route53_zone.selected.*.zone_id[0]
   name    = var.hostnames[count.index]
   type    = "CNAME"
   ttl     = "300"
   records = list(var.alb_dns_name)
 }
+
+data "aws_lb" "alb_selected" {
+  count = var.alb_only && var.hostname_create && length(var.hostnames) != 0 ? length(var.hostnames) : 0
```

Comment:
I think need to add the condition `var.hosted_zone_is_internal` here, otherwise this data resource will be created even if the hostname is not internal.
Also rename from `alb_selected` to `alb_internal_selected`

Reply:
Done, @adenot !